tests/test_data/test_structs.py

```diff
@@ -1,5 +1,109 @@
+"""
+MockResponse: Functions like a regular Response from Requests
+MockResponseInternal: Responses which are accessed in within handle_peer_requests
+    have already been modified by the time they are received.
+"""
+
+
+class MockResponse:
+    def __init__(self, json_data, status_code):
+        self.json_data = json_data
+        self.status_code = status_code
+
+    def json(self):
+        return self.json_data
+
+    def status_code(self):
+        return self.status_code
+
+
+# Internal Response needs a result() function since it's supposed to be a Future response
+
+class MockResponseInternal:
+    def __init__(self, json_data, status_code):
+        self.json_data = json_data
+        self.status_code = status_code
+
+    def json(self):
+        return {"results": self.json_data,
+                "status": self.status_code}
+
+    def result(self):
+        return MockResponseInternal(self.json_data, self.status_code)
+
+
+class testHeader():
+    def __init__(self, headers):
+        self.headers = headers
+
+    def headers(self):
+        return self.headers
+
+
+TWO = {
+    "p1": "http://10.9.208.132:6000",
+    "p2": "http://10.9.208.132:8000"
+}
+THREE = {
+    "p1": "http://10.9.208.132:6000",
+    "p2": "http://10.9.208.132:8000",
+    "p3": "http://10.9.208.132:9000"
+}
+
+
+exampleHeaders = testHeader({
+        "Content-Type": "application/json",
+        "Host": "ga4ghdev01.bcgsc.ca:8890",
+        "User-Agent": "python-requests/2.22.0",
+        "Accept": "application/json",
+        "Accept-Encoding": "gzip, deflate",
+        "Authorization": "Bearer eyJhbGciOiJSUzI1NiIsI",
+        "Federation": "false"
```

Comment:
what happens when you specify False or True in the federation header? we might have discussed on this but I forgot what the result was

Reply:
Added a lower() call to the check to catch all variations of "false." Anything else will federate.

Reply:
Ok, cool